src/QirRuntime/test/QIR-static/qir-test-conditionals.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+#include "catch.hpp"
+
+#include "CoreTypes.hpp"
+#include "QirContext.hpp"
+#include "QirTypes.hpp"
+#include "QuantumApi_I.hpp"
+#include "SimulatorStub.hpp"
+
+using namespace std;
+using namespace Microsoft::Quantum;
+
+// TestConditionalOnResult() is authored in a way that the expected path through the function only applies X operator
+// for the chosen sequence of measurement results, and all other paths apply Y. Thus, the correct execution must get the
+// expected maximum number of X and ControlledX callbacks.
+struct ConditionalsTestSimulator : public Microsoft::Quantum::SimulatorStub
+{
+    int cX = 0;
+    int ccX = 0;
+
+    vector<ResultValue> mockMeasurements;
+    int nMeasure = -1;
```

Comment:
FYI: I suspect you meant `0` instead of `-1`.

Reply:
No, I do mean -1 here as I increment first thing the Measure callback. Given that you've found it confusing, I'll change the code to start from 0.